python-interface/src/MiniBotFramework/Communication/ZMQ.py

```diff
@@ -115,41 +153,51 @@ def receive(self, receivedQueue = None):
         """
         oldData = "empty"
         while True:
-            # wait infinitely to receive the message
-            if self.sub.poll(timeout=0):
-                data = self.sub.recv_multipart()
-                #print "received ", data
-                
-                if oldData != data:
-                    # parse the data into lWheel and rWheel and send it as a
-                    # tuple
-                    start = data[1].find("(")
-                    comma = data[1].find(",")
-                    end = data[1].find(")")
-                    lWheel = int(data[1][start + 1:comma])
-                    rWheel = int(data[1].strip()[comma + 1:end])
-                    info = (lWheel, rWheel)
-    
-                    # do something, send commands
-                    if receivedQueue is not None:
-                        receivedQueue.put(info)
-                    else:
-                        #print "received ", info
-                        pass
-                    oldData = data
-        
+            if self.isReceiver:
+                try:
+                    # wait infinitely to receive the message
+                    if self.sub.poll(timeout=0):
+                        data = self.sub.recv_multipart()
+                        #print "received ", data
+                        
+                        if oldData != data:
+                            # parse the data into lWheel and rWheel and send it as a
+                            # tuple
+                            start = data[1].find("(")
+                            comma = data[1].find(",")
+                            end = data[1].find(")")
+                            lWheel = int(data[1][start + 1:comma])
+                            rWheel = int(data[1].strip()[comma + 1:end])
+                            info = (lWheel, rWheel)
```

Comment:
we want to send tuples from the frontend to zmqsend, and get tuples from zmqreceive for uniformity. let zmq understand interpret all those things